document/workflow/variable-tree/data-source/skeleton-tree-node.ts

```diff
@@ -0,0 +1,487 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ */
+import { FullQualifiedName, RuntimeContext, Xo, XoJson, XoStructureArray, XoStructureField, XoStructureObject, XoStructurePrimitive, XoStructureType } from '@zeta/api';
+import { GraphicallyRepresented } from '@zeta/base';
+import { BehaviorSubject, Observable, filter, first, forkJoin, map, of, switchMap, tap } from 'rxjs';
+import { Draggable } from '../../shared/drag-and-drop/mod-drag-and-drop.service';
+import { RecursiveStructurePart } from '@pmod/xo/expressions/comparable-path';
+import { TreeNodeFactory, TreeNodeObserver } from './skeleton-tree-data-source';
+
+export abstract class SkeletonTreeNode implements GraphicallyRepresented<Element>, Draggable {
+    private _structure: XoStructureField;
+    private _xfl: string;
+    private _isList: boolean;
+    collapsible = true;
+    private _collapsed = true;
+
+    /** Node is marked for some reason and can be rendered differently than an unmarked node */
+    private readonly _marked$ = new BehaviorSubject<boolean>(false);
+    private readonly _selected$ = new BehaviorSubject<boolean>(false);
+
+    private readonly _graphicalRepresentation$ = new BehaviorSubject<Element>(null);
+
+    protected _children: SkeletonTreeNode[] = [];
+    protected _parent: SkeletonTreeNode;
+
+    constructor(structure: XoStructureField, protected nodeFactory: TreeNodeFactory, protected nodeObservers: Set<TreeNodeObserver> = new Set<TreeNodeObserver>()) {
+        this.setStructure(structure);
+    }
+
+
+    getStructure(): XoStructureField {
+        return this._structure;
+    }
+
+
+    protected setStructure(structure: XoStructureField) {
+        this._structure = structure;
+    }
+
+
+    protected changeStructureType(type: XoStructureType) {
+        this._structure.typeFqn = type.typeFqn;
+        this._structure.typeRtc = type.typeRtc;
+        this._structure.typeLabel = type.typeLabel;
+    }
+
+
+    get collapsed(): boolean {
+        return this._collapsed;
+    }
+
+
+    collapse() {
+        if (!this.collapsed) {
+            this._collapsed = true;
+            this.notifyObservers();
+        }
+    }
+
+
+    uncollapse() {
+        if (this.collapsible && this.collapsed) {
+            this._collapsed = false;
+            this.notifyObservers();
+        }
+    }
+
+
+    uncollapseRecursivelyUpwards() {
+        this.parent?.uncollapseRecursivelyUpwards();
```

Comment:
If you want, that this node appear in the tree, then you need to uncollapse all parents recursly, and you dont know, if this is uncollapsable, but evary parent should be uncollapsable, because they have at leased one parent.
The solution of https://github.com/Xyna-Factory/xyna-process-modeller/issues/90 makes this function obsolete.